feat: ✨ sd-video

abudd1094 opened the pull request:
adds sd-video component: https://github.com/solid-design-system/solid/issues/281

Please see my comment on the ticket above for status information

mariohamann commented on packages/components/src/components/video/videoExample.ts:
```diff
@@ -0,0 +1,75 @@
+    return html`
+      <div>
+        <sd-video ?playing=${this.playing} ?overlay=${this.overlay}>
+          <video
+            @play="${this.play}"
+            @pause="${this.pause}"
+            controls
+            poster="https://www.blender.org/wp-content/uploads/2020/10/robin-tran-redautumnforest_pr1.jpg"
+            style="max-width: 854px; may-height: 480px;"
+          >
+            <source src="http://media.w3.org/2010/05/sintel/trailer.mp4" type="video/mp4" />
+            Your browser does not support the video tag.
+          </video>
+          <sd-icon
+            library="global-resources"
+            name="system/start"
+            slot="play-icon"
+            color="primary"
+            style="font-size: 64px; transform: translateX(5px);"
+          ></sd-icon>
+        </sd-video>
```
This should be part of a sample. This whole file shouldn't exist, as this is not a component.

In the sample We should wire up plain JS with the component.

abudd1094 replied:
is there any other example in the repo to look to?

mariohamann commented on packages/components/src/components/video/video.ts:
```diff
@@ -0,0 +1,77 @@
+            'absolute left-[50%] top-[50%] translate-x-[-50%] translate-y-[-50%] p-4 bg-white bg-opacity-75 rounded-full flex items-center justify-center transition-opacity duration-300'
+          )}
+        >
+          <slot name="play-icon" part="play-icon"></slot>
```
The play icon should become a system icon and set as default slot content here.

abudd1094 replied:
What does "become a system icon" mean?

abudd1094 replied:
I assume you want:
`<sd-icon library="global-resources" name="system/start" color="primary" class="text-[4rem]"></sd-icon>`

To become something like:
`<sd-icon library="system" name="start" color="primary" class="text-[4rem]"></sd-icon>`

But that doesn't work.  Please provide further instructions

mariohamann replied:
As soon as you merge main, it should work now as @Vahid1919 just merged an update of https://github.com/solid-design-system/solid/blob/main/packages/components/src/components/icon/library.system.ts in his latest commit to main.

mariohamann commented on packages/components/src/components/video/videoExample.ts:
```diff
@@ -0,0 +1,75 @@
+            <source src="http://media.w3.org/2010/05/sintel/trailer.mp4" type="video/mp4" />
+            Your browser does not support the video tag.
+          </video>
+          <sd-icon
+            library="global-resources"
+            name="system/start"
+            slot="play-icon"
+            color="primary"
+            style="font-size: 64px; transform: translateX(5px);"
+          ></sd-icon>
```
Please remove.

abudd1094 replied:
remove what exactly?  This whole component will be removed as per the comment above

karlbaumhauer commented on packages/components/src/components/video/video.stories.ts:
```diff
@@ -0,0 +1,79 @@
+ * The `sd-video` component can be used to wrap external video elements in order to provide basic styling for Union Investment.
+ * ## Implementation
+ * Use in combination with a viewer component (e. g. from Moving Image) or a bare `<video>`-Tag in addition to `sd-media`.
+ * `<sd-media>
```
should we add some information here about the non-existing sd-media component? Or maybe leave it out of here and add it as soon as the component is ready.

mariohamann replied:
+1 – would remove everything related `sd-media`

karlbaumhauer commented on packages/components/src/components/video/video.ts:
```diff
@@ -0,0 +1,86 @@
+
+/**
+ * @summary Used to wrap external video elements (e. g. from Moving Image or bare <video>-Tags) and provide some basic styling for Union Investment.
+ * @status experimental
```
pls change before merging

mariohamann replied:
You mean the `status`?

abudd1094 replied:
what should status be?

karlbaumhauer replied:
status and version.

Status should be stable and version needs to be updated to the latest +1 right before merging.

mariohamann commented:
@solid-design-system/design 

I know, that the play button is mathematically correct centered but it feels visually off.

<img width="853" alt="CleanShot 2023-10-11 at 09 18 34@2x" src="https://github.com/solid-design-system/solid/assets/26542182/6e27af31-93ec-40d7-a3e5-166beec2474e">

I would propose to add a 4px margin to the left to make it visually more pleasing:

<img width="853" alt="CleanShot 2023-10-11 at 09 20 11@2x" src="https://github.com/solid-design-system/solid/assets/26542182/fb338dfc-0a3c-4e22-baa2-c5fcbe05859d">

Apparently in Figma there already is a little margin to the left in the icon itself:

<img width="432" alt="CleanShot 2023-10-11 at 09 22 55@2x" src="https://github.com/solid-design-system/solid/assets/26542182/d8cd93c9-46e7-49de-81df-5f589da664f2">

@Vahid1919 Do you remember, where the icon was coming from?

mariohamann commented:
The default story should contain an example where you can play around with the knobs. We should add a placeholder image in format 16:9 for that in the background. 

I just opened a PR for the image: https://github.com/solid-design-system/solid/pull/457

van-nguyen-ht commented:
@mariohamann yes indeed, in [figma](https://www.figma.com/file/VTztxQ5pWG7ARg8hCX6PfR/Solid-DS-%E2%80%93-Component-Library?type=design&node-id=1663-18047&mode=design&t=YPSuBz0TT42I9sPK-11) the icon has a padding to the left

mariohamann commented on packages/components/src/components/video/video.stories.ts:
```diff
@@ -0,0 +1,79 @@
+  decorators: [withActions] as any
+};
+
+/**
+ * `sd-video` does not have any content by default because it is a wrapper component.
+ */
+
+export const Default = {
+  render: (args: any) => {
+    return generateTemplate({ args });
+  }
+};
```
As mentioned here https://github.com/solid-design-system/solid/pull/451#issuecomment-1757026290 we should show at least an image in the main slot.

mariohamann commented on packages/components/src/components/video/video.ts:
```diff
@@ -0,0 +1,86 @@
+  /** Show a darker overlay. */
+  @property({ type: Boolean, reflect: true }) overlay = false;
+
+  play() {
+    this.emit('sd-play');
+    this.playing = true;
+  }
```
If we provide this, we should have a second method, that does the opposite.

abudd1094 replied:
I don't think it makes sense to expose this or add a "pause" method, control is always coming from the wrapped component and should be reflected via the property.  

I propose marking this function as private and using only for the click event to make it simple.  External control should use the "playing" property exclusively.

karlbaumhauer replied:
good point... @mariohamann what do you think?

mariohamann replied:
Ok. Either both or none. In accordion we provide it like this:

<img width="1050" alt="CleanShot 2023-10-11 at 13 23 20@2x" src="https://github.com/solid-design-system/solid/assets/26542182/cfa11ca3-e625-4270-b094-527d7995f532">

But I think it's totally valid to just work with the prop here.

mariohamann commented on packages/components/src/components/video/video.ts:
```diff
@@ -0,0 +1,86 @@
+ * @event sd-play - Event emitted when the video is played or paused. Detail reflects "playing" property.
+ *
+ * @slot - The default slot used to pass a video player element.
+ * @slot play-icon - The video's play icon.
```
❤️

mariohamann commented on packages/components/src/components/video/video.ts:
```diff
@@ -0,0 +1,86 @@
+export default class SdVideo extends SolidElement {
+  private readonly hasSlotController = new HasSlotController(this, '[default]', 'play-icon');
+
+  /** True if the contained video is playing. Hides everything when true. */
```
```suggestion
  /** Set to `true` to hide the play icon and the overlay. */
```

mariohamann commented on packages/components/src/components/video/video.ts:
```diff
@@ -0,0 +1,86 @@
+  /** True if the contained video is playing. Hides everything when true. */
+  @property({ type: Boolean, reflect: true }) playing = false;
+
+  /** Show a darker overlay. */
```
```suggestion
  /** Set to `true` to show a dark overlay. Only used when `playing` is `false`. */
```

mariohamann commented on packages/components/src/components/video/video.ts:
```diff
@@ -0,0 +1,86 @@
+          @click=${this.play}
+          class=${cx(
+            this.playing ? 'opacity-0' : 'opacity-100',
+            'absolute left-[50%] top-[50%] translate-x-[-50%] translate-y-[-50%] p-4 bg-white bg-opacity-75 rounded-full flex items-center justify-center transition-opacity duration-300'
```
"transition" === 😍

mariohamann commented on packages/components/src/components/video/video.stories.ts:
```diff
@@ -0,0 +1,79 @@
+    return generateTemplate({ args });
+  }
+};
+
```
When implementing the placeholder, make another story which does `Playing x Overlay`.

abudd1094 commented:
> @solid-design-system/design
> 
> I know, that the play button is mathematically correct centered but it feels visually off.
> 
> <img alt="CleanShot 2023-10-11 at 09 18 34@2x" width="853" src="https://user-images.githubusercontent.com/26542182/274176037-6e27af31-93ec-40d7-a3e5-166beec2474e.png">
> I would propose to add a 4px margin to the left to make it visually more pleasing:
> 
> <img alt="CleanShot 2023-10-11 at 09 20 11@2x" width="853" src="https://user-images.githubusercontent.com/26542182/274176184-fb338dfc-0a3c-4e22-baa2-c5fcbe05859d.png">
> Apparently in Figma there already is a little margin to the left in the icon itself:
> 
> <img alt="CleanShot 2023-10-11 at 09 22 55@2x" width="432" src="https://user-images.githubusercontent.com/26542182/274176967-d8cd93c9-46e7-49de-81df-5f589da664f2.png">
> @Vahid1919 Do you remember, where the icon was coming from?

I had a translate on there I thought you said to "remove".  Margin will distort the background:
![image](https://github.com/solid-design-system/solid/assets/39494579/c4afec6e-f3fc-447c-a236-fc04adb89a37)

I will re-add the translate

karlbaumhauer commented on packages/components/src/components/video/video.ts:
```diff
@@ -7,12 +7,12 @@ import SolidElement from '../../internal/solid-element';
- * @status experimental
- * @since 1.0
+ * @status stable
+ * @since 1.16.0
```
```suggestion
 * @since 1.19.0
```

Vahid1919 commented:
> @Vahid1919 Do you remember, where the icon was coming from?

It was from the icon-library from the previous Union Investment project I believe.

mariohamann commented:
> > @Vahid1919 Do you remember, where the icon was coming from?
> 
> It was from the icon-library from the previous Union Investment project I believe.

It seems like the icon is not correctly set in it's frame. Could you please add a ticket, that we have to include its padding and remove the added translate by Alec then?

mariohamann commented:
I could get the video playing when doing it via keyboard.

https://64469bd84c5ea1ce2dd5d016-nziojohwnh.chromatic.com/?path=/story/components-sd-video--video-element

mariohamann commented:
So we should have one default story with jsut the imag e + one Overlay x Playing Story. @abudd1094  :)

Vahid1919 commented:
I noticed that the default play button is still visible underneath our own for Firefox --> 
![image](https://github.com/solid-design-system/solid/assets/63849626/6d5c383c-052d-48a7-82ae-b512137a89b8)

PS: This only seems to be the case for the first button press, it does not come back afterwards.

abudd1094 commented:
> I could get the video playing when doing it via keyboard.
> 
> https://64469bd84c5ea1ce2dd5d016-nziojohwnh.chromatic.com/?path=/story/components-sd-video--video-element
> 
> And I'm not sure regarding the focus styles – I think we should use our default ones.

By "default" you mean: focus-visible:focus-outline

mariohamann commented:
@karlbaumhauer aybe we should reintroduce the `poster` slot – this would prevent everything here?

karlbaumhauer commented:
> @karlbaumhauer aybe we should reintroduce the `poster` slot – this would prevent everything here as it could just be an overlay over the video?

thats a good idea...

abudd1094 commented:
@karlbaumhauer 

- I removed my translate for the icon as requested, Vahid's PR should fix that without further changes in the sd-video.
- The play icon is centered in our component, which does not take into account the "controls" in the underlying video component.  

I'll add a poster slot, we'll have a quick check in the daily, and I'll add stories / tests if that is accepted.

mariohamann commented:
@abudd1094 @karlbaumhauer  An alternative could be to only show controls when playing. This could work when you extend your JS, wouldn't it?

abudd1094 commented:
@karlbaumhauer @mariohamann 

OK, after much pain and learning, sd-video is ready for another review.

I have added a "poster" slot which seeks to mirror the behavior of the "poster" attribute on the native "video" element.  It will be shown under the optional overlay and on top of the default video element slot with no pointer events.  So the user must click (or use keyboard to activate) the "play-icon" for the initial play, after which the "poster" will be hidden, and the user can click through the overlay to the video controls.  

I have added as many tests as I could think to add.  I would also like to add the basic "accessibility" test, but I'm not sure I understand how it works.  It seems to fail for Chrome here but I cannot figure out the error:
![image](https://github.com/solid-design-system/solid/assets/39494579/84085bdf-173c-4bcb-93f6-7ed4d6faaa3f)

Accessibility has been further bolstered and I added a story for "mouseless" control.  Please advise.

karlbaumhauer commented:
@abudd1094 sorry for the delay... I will have a look into it by tomorrow.

karlbaumhauer commented on packages/components/src/components/video/video.stories.ts:
```diff
@@ -21,21 +36,109 @@ export default {
+// Script used in next 2 stories
+const videoElementScript = html`<script>
+  function runScriptOnPathChange() {
+    // *NOTE: Storybook single page behavior complicates queries between the "Docs" and "Video Element" stories.
```
I like that one...!

karlbaumhauer commented on packages/components/src/components/video/video.stories.ts:
```diff
@@ -21,21 +36,109 @@ export default {
 /**
- * Use in combination with a viewer component (e. g. from Moving Image) or a bare `<video>`-Tag. Here we wrap a `video` element that includes a `poster` attribute.
+ * Use in combination with a viewer component (e. g. from Moving Image) or a bare `<video>`-Tag. <br/>
+ * Here we wrap a `video` element that includes a `poster` attribute.
```
```suggestion
 * Here we wrap a `video` element that includes setting an image to the `poster` slot.
```

karlbaumhauer commented on packages/components/src/components/video/video.stories.ts:
```diff
@@ -47,31 +150,45 @@ export const VideoElement = {
 };
+
+/**
+ * Duplicates the previous story with default focus on the play button.
```
we need to deactivate the mouse in this story, when hovering over the component. Have a look into the sd-tag mouseless story to see how its implemented.

karlbaumhauer commented:
> @karlbaumhauer @mariohamann
> 
> OK, sd-video is finally ready for another review.
> 
> I have added a "poster" slot which seeks to mirror the behavior of the "poster" attribute on the native "video" element. It will be shown under the optional overlay and on top of the default video element slot with no pointer events. So the user must click (or use keyboard to activate) the "play-icon" for the initial play, after which the "poster" will be hidden, and the user can click through the overlay to the video controls.
> 
> I have added as many tests as I could think to add. I would also like to add the basic "accessibility" test, but I'm not sure I understand how it works. It seems to fail for Chrome here but I cannot figure out the error:
> 
> Accessibility has been further bolstered and I added a story for "mouseless" control. Please advise.

So the component looks good and I just found some small text changes and 1 issue in the mouseless story.

Regarding the tests, I unfortunately cannot help here. @Vahid1919 or @mariohamann pls advise if possible.

mariohamann commented:
<img width="631" alt="CleanShot 2023-10-18 at 10 18 00@2x" src="https://github.com/solid-design-system/solid/assets/26542182/9a0e806e-e179-4a57-a4a2-938b3619ec00">

Could you please set the size on the image to prevent this?

I think you're good to go with `class="w-WHATEVER-WIDTH aspect-video cover"`.

mariohamann commented:
Please use one of our internal images as poster. As these will be used for screenshot tests, we shouldn't have any external dependencies on them. (That's why we e. g. intercept all calls to global-resources and provide own icons, as we don't want to have any external stuff.) 

<img width="959" alt="CleanShot 2023-10-18 at 10 20 47@2x" src="https://github.com/solid-design-system/solid/assets/26542182/1d2afce4-0bd6-496b-bd5f-7762f21d9704">

mariohamann commented on packages/components/src/components/video/video.stories.ts:
```diff
@@ -0,0 +1,194 @@
+  type: 'slot',
+  name: 'default',
+  value:
+    '<video controls id="videoExample" style="min-width: 854px; min-height: 480px; width: 854px; height: 480px;"><source src="http://media.w3.org/2010/05/sintel/trailer.mp4" type="video/mp4" />Your browser does not support the video tag.</video>'
```
You can use here Tailwind as well. Something like `w-ANYWIDTH aspect-video cover`.

mariohamann commented on packages/components/src/components/video/video.stories.ts:
```diff
@@ -0,0 +1,194 @@
+      document.querySelectorAll('#sdVideoExample > sd-video')[1] ||
+      document.querySelectorAll('#sdVideoExample > sd-video')[0];
+
+    function setupEventListeners() {
+      sdVideo?.addEventListener('sd-play', playVideo);
+      videoEl?.addEventListener('play', updatePlayingStatus);
+      videoEl?.addEventListener('pause', updatePlayingStatus);
+    }
+
+    function playVideo() {
+      videoEl.play();
+    }
+
+    function updatePlayingStatus() {
+      sdVideo.playing = videoEl.paused ? false : true;
+    }
+
+    function removeEventListeners() {
+      // ensure we affect ANY existing elements by querying for all, see *NOTE above
+      const videoEls = document.querySelectorAll('#videoExample');
+      const sdVideos = document.querySelectorAll('#sdVideoExample > sd-video');
+      const sdVideoPlayButtons = document.querySelectorAll('#sdVideoExample > sd-video');
```
Could you add some more documentation for the users of our library?

Pointing them to what's important for them etc. Some stuff isn't important, like Storybook changes, maybe you can clarify that, too.

So more thinking from a user of our library: What is importatnt. What do I have to do why. What's not important.

abudd1094 replied:
You mean via code comments in this block?

mariohamann replied:
Yes

mariohamann commented on packages/components/src/components/video/video.stories.ts:
```diff
@@ -0,0 +1,194 @@
+  },
+  render: (args: any) => {
+    return html`
+      <div id="sdVideoExample">
```
```suggestion
      <div id="sd-video-example">
```

mariohamann commented on packages/components/src/components/video/video.test.ts:
```diff
@@ -0,0 +1,137 @@
+const posterSlot = html`<img
+  slot="poster"
+  alt="poster"
+  src="https://www.blender.org/wp-content/uploads/2020/10/robin-tran-redautumnforest_pr1.jpg"
```
Especially here: Definitely remove external dependencies. If needed just inline the image with base64 or whatever.

abudd1094 replied:
"just inline the image with base64 or whatever" --- please expand on what you mean.  I don't think you want:
![image](https://github.com/solid-design-system/solid/assets/39494579/507211f3-205b-4e6d-abba-e34989ae0cfc)
I have removed the src completely, it still passes the tests.  The video remains in the tests and is an external dependency.  If you have a better way to do it, please explain, otherwise I'll leave it like this

mariohamann commented on packages/components/src/components/video/video.ts:
```diff
@@ -0,0 +1,137 @@
+          role="presentation"
+          class=${cx(
+            this.overlay && !this.playing ? 'opacity-100' : 'opacity-0',
+            'bg-[rgba(0,0,0,0.65)] w-full h-full absolute top-0 left-0 pointer-events-none z-20 play-pause-transition'
```
```suggestion
            'bg-black/65 w-full h-full absolute top-0 left-0 pointer-events-none z-20 play-pause-transition'
```

Couldn't you use something like bg-black/65 or something?

abudd1094 replied:
discussed via phone --- no bg-black token exists :(

mariohamann commented on packages/components/src/components/video/video.ts:
```diff
@@ -0,0 +1,137 @@
+          @keydown=${this.play}
+          class=${cx(
+            this.playing ? 'opacity-0 pointer-events-none' : 'opacity-100',
+            'absolute left-[50%] top-[50%] translate-x-[-50%] translate-y-[-50%] p-4 bg-white bg-opacity-75 rounded-full flex items-center justify-center z-30 play-pause-transition'
```
```suggestion
            'absolute left-1/2 top-1/2 translate-x-1/2 translate-y-1/2 p-4 bg-white bg-opacity-75 rounded-full flex items-center justify-center z-30 play-pause-transition'
```

Something like this should work.

abudd1094 replied:
Before:
![image](https://github.com/solid-design-system/solid/assets/39494579/7cf3b09d-4a8c-4ac4-b429-f51fc5110e9a)

After:
![image](https://github.com/solid-design-system/solid/assets/39494579/58c9aa25-7333-4033-9978-f5b0ab025cbf)

mariohamann commented:
> > Please use one of our internal images from our placeholders package as poster. As these will be used for screenshot tests, we shouldn't have any external dependencies on them. (That's why we e. g. intercept all calls to global-resources and provide own icons, as we don't want to have any external stuff.)
> > <img alt="CleanShot 2023-10-18 at 10 20 47@2x" width="959" src="https://user-images.githubusercontent.com/26542182/276168241-1d2afce4-0bd6-496b-bd5f-7762f21d9704.png">
> 
> What about for the video?

As this doesn't show up in the screenshots, I'm less opinionated there. We maybe should have a default there, too. But for the moment I think that's fine.

abudd1094 commented:
All comments have been addressed.  @mariohamann I added this issue regarding the accessibility test:
https://github.com/solid-design-system/solid/issues/483

Please let me know if I should refine this ticket or make further changes.

karlbaumhauer commented on packages/components/src/components/video/video.stories.ts:
```diff
@@ -13,13 +13,18 @@ const videoConstant: ConstantDefinition = {
   name: 'poster',
   value:
-    '<img slot="poster" alt="poster" src="https://www.blender.org/wp-content/uploads/2020/10/robin-tran-redautumnforest_pr1.jpg" />'
+    '<img slot="poster" alt="poster" class="w-WHATEVER-WIDTH aspect-video cover" src="./placeholders/architecture.jpg" />'
```
@mariohamann was this `w-WHATEVER-WIDTH` really what you expected? If yes, why? Would love to then see a description here.

karlbaumhauer commented:
> All comments have been addressed. @mariohamann I added this issue regarding the accessibility test: #483
> 
> Please let me know if I should refine this ticket or make further changes.

@abudd1094 I looked at the ticket and re-ordered it a bit. thx for creating it.

abudd1094 commented:
@van-nguyen-ht 
![image](https://github.com/solid-design-system/solid/assets/39494579/7df5c7ee-bc74-4b58-8343-c900be876492)

Point 1:
It is pixel for pixel the same, 96 x 96, please check the measurements by number:
![image](https://github.com/solid-design-system/solid/assets/39494579/a0d06ab7-978f-4dde-bd2e-f2e6d661e631)
![image](https://github.com/solid-design-system/solid/assets/39494579/0295d650-5146-4b06-96c0-13149c1b44ba)

Point 2:
"There's no interactions with the play button" --- yes, that is expected.  In the default story, the component uses a placeholder image as its "default slot" so there is no video to play.  Please see the Video Element stories for interaction demonstrations.

coraliefeil commented:
SRY messed up assignees by accident

abudd1094 commented:
@van-nguyen-ht requested that I implement the queries from the design:
![image](https://github.com/solid-design-system/solid/assets/39494579/e4499725-c693-4182-b430-a4db63bcb9a1)

@mariohamann I didn't see 414px as a standard breakpoint, it is implemented in the css block.  If there is a more desirable way to implement this or you want to adjust the format please advise.

EllaELSA commented:
Not sure where to add a comment, here or in chromatic? 
Shouldn't we provide a hover effect on video or at least at the play button?

coraliefeil commented:
Removed myself ... Van’s taking care of design review

coraliefeil commented:
@abudd1094 could we use the sd-interactive for the play icon and the close icon? states are the same.

abudd1094 commented:
@EllaELSA 
This appears to be a different element entirely <sd-play-button> which is not mentioned in the ticket.

@coraliefeil 
What is sd-interactive?  I don't see it in Storybook or Figma... I do see a ticket in Review, perhaps it is not yet merged then.  In that case, we can use it, but I'll have to wait until it is merged.  

@mariohamann @karlbaumhauer @yoezlem 
Seems like we missed a dependency.  Should I add <sd-play-button> and include it in the default slot as part of this PR?

Sounds like what we want, is to merge sd-interactive, use it to build the sd-play-button, and then use that as the default slot for sd-video???

That is fine for me but this information was not transparent.  Please advise before I continue.

coraliefeil commented:
@abudd1094 https://solid-design-system.fe.union-investment.de/x.x.x/storybook/?path=/docs/styles-sd-interactive--docs

mariohamann commented:
I'm not sure if the `sd-interactive` class already works inside components. Maybe it does, maybe not. I would suggest du create a follow-up ticket for a possible refactoring, but as long as the current styles are fine, we should proceed.

mariohamann commented:
@abudd1094 You can check an implemetnation example in sd-carousel:

import InteractiveStyles from '../../styles/interactive/interactive.css?inline';

You can sk @Vahid1919 for further implementation details.

mariohamann commented:
@abudd1094 You could either set `poster` and `overlay` to pointer-events-none, so that clicking will be passed to the element or you set the click handler on the whole element.

@EllaELSA is right, that clicking anywhere should trigger playing: https://www.w3schools.com/html/html5_video.asp https://www.w3schools.com/tags/att_video_poster.asp

Besides that: Hover styles aren't defined anywhere for cover elements (as this isn't defined in design) and doesn't work in current UI videos as well. I suggest this to be an improvement if design catched up with everything.

mariohamann commented:
@abudd1094 I don't understand the purpose of the `sd-play-button` as well, especially as should never handle pause with a big filling icon on the video?

I would definitely say to skip this component here and we will validate, if there's a reason for the `sd-play-button`. If yes, we might refactor at a later point, but for the moment, I don't see anything we can benefit from it.

abudd1094 commented:
@mariohamann @karlbaumhauer @EllaELSA @coraliefeil @van-nguyen-ht 
Play button now contains "sd-interactive", covers the whole element to field all click events and applies hover style to the play icon.  Click events have been fine tuned for precision.  I added a debounce functionality in the story to prevent flickering of our play icon on "seeking" events, without this, the icon will show up briefly every time a user initiates seeking in the video player.

abudd1094 commented:
77 comments, I'm going for a record...

@van-nguyen-ht @mariohamann 
I moved the media query from screen to element as mentioned on Figma.  This should cover all of the feedback we have gathered here.  Will update PR again when Quality Gate is fixed.

mariohamann commented on packages/components/src/components/navigation-item/navigation-item.ts:
```diff
@@ -10,8 +10,8 @@ import SolidElement from '../../internal/solid-element';
  * @summary Flexible button / link component that can be used to quickly build navigations. Takes one of 3 forms: link (overrides all other if 'href' is provided), button (default), or accordion (if 'children' slot defined).
- * @status experimental
- * @since 1.12.0
+ * @status stable
+ * @since 1.15.0
```
Is this intended?

abudd1094 replied:
yes, the version was incorrect --- regarding status, it is not clear to me how you determine that, imo this is now stable and not experimental

karlbaumhauer replied:
everything is stable unless we specifically announce something as experimental

abudd1094 commented:
@van-nguyen-ht @coraliefeil @EllaELSA 
I have added you as assignees for the UI Tests at your lesiure.

@Vahid1919 @yoezlem @DanielHargesheimer 
I have adde you as assignees simply to have a look at this new feature (as discussed this morning in the daily).  No need to review!
